.github/workflows/image-digest-updater-main.yml

```diff
@@ -0,0 +1,71 @@
+name: Image Digest Updater (Reusable)
+on:
+  workflow_call:
+    inputs:
+      update-args:
+        description: 'Arguments to pass to update.sh (e.g., --components or --exclude-components)'
+        required: false
+        type: string
+        default: ''
+      job-name:
+        description: 'Name for the update job'
+        required: false
+        type: string
+        default: 'update-images'
+    secrets:
+      AZURE_CLIENT_ID:
+        description: 'Azure Client ID'
+        required: true
+      AZURE_TENANT_ID:
+        description: 'Azure Tenant ID'
+        required: true
+      AZURE_SUBSCRIPTION_ID:
+        description: 'Azure Subscription ID'
+        required: true
+env:
+  AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+  AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+  AZURE_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+permissions:
+  contents: write
+  pull-requests: write
+jobs:
+  update-images:
+    name: ${{ inputs.job-name }}
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        fetch-depth: 0
+    - name: 'Set up Go'
+      uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+      with:
+        go-version-file: 'go.work'
+        check-latest: true
+    - name: Configure Git
+      run: |
+        git config --global user.name "github-actions[bot]"
+        git config --global user.email "github-actions[bot]@users.noreply.github.com"
+    - name: Set up GitHub CLI
+      run: |
+        echo "${{ secrets.GITHUB_TOKEN }}" | gh auth login --with-token
+    - name: Install Azure CLI
+      uses: Azure/ARO-HCP@main
+    - name: Azure CLI login
+      uses: azure/login@v2
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+    - name: Azure Container Registry login
+      run: |
+        # Login to dev ACR (used for images in config)
+        az acr login --name arohcpsvcdev || echo "Warning: Could not login to arohcpsvcdev ACR"
```

Comment:
This can result in PRs with missing image updates if unable to login to arohcpsvcdev. I think we should fail fast if login fails and optionally adding retry logic.